File: atcoder.jp/contests/m-solutions2020/tasks/m_solutions2020_b/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

let [a, b, c] = input().split(" ").map(Number);
const K = +input();

for (let i = 0; i <= K; i++) {
  if (a < b && b < c) {
    console.log("Yes");
    process.exit();
  }

  if (a >= b) {
    b *= 2;
  } else if (b >= c) {
    c *= 2;
  }
}

console.log("No");

File: atcoder.jp/contests/m-solutions2020/tasks/m_solutions2020_b/main.rs
use std::process;

use proconio::input;

fn main() {
    input! {
        a: usize,
        mut b: usize,
        mut c: usize,
        k: usize,
    }

    for _ in 0..=k {
        if a < b && b < c {
            println!("Yes");
            process::exit(0);
        }

        if a >= b {
            b *= 2;
        } else if b >= c {
            c *= 2;
        }
    }

    println!("No");
}
